src/Model.jl

```diff
@@ -118,9 +118,10 @@ function Model(lattice::AbstractMatrix{T},
         if !disable_electrostatics_check && any(!iszero, charge_ionic.(atoms))
             error("Coulomb electrostatics is incompatible with fixed Fermi level.")
         end
-    else # fixed number of electrons
+    else  # fixed number of electrons
         n_electrons < 0 && error("n_electrons should be non-negative.")
-        if !disable_electrostatics_check && n_electrons_from_atoms(atoms) != n_electrons
+        if !disable_electrostatics_check && any(!iszero, charge_ionic.(atoms)) &&
```

Comment:
This shenanigans is the reason why I initialised `disable_electrostatics_check=all(iszero, charge_ionic.(atoms))` in my version of this PR. I suggest we do the same here to simplify this logic.